src/context/SyntheticsStateContext/selectors/tradeSelectors.ts

```diff
@@ -133,15 +133,29 @@ export const makeSelectFindSwapPath = createSelectorFactory(
       const allPaths = q(selectAllPaths);
       const estimator = q(selectSwapEstimator);
 
-      const findSwapPath: FindSwapPath = (usdIn: bigint, opts: { byLiquidity?: boolean }) => {
+      const findSwapPath: FindSwapPath = (usdIn: bigint, opts: { order?: ("liquidity" | "length")[] }) => {
         if (!allPaths?.length || !estimator || !marketsInfoData || !fromTokenAddress) {
           return undefined;
         }
 
         let swapPath: string[] | undefined = undefined;
-
-        if (opts.byLiquidity) {
-          swapPath = allPaths[0].path;
+        const sortedPaths = opts.order
+          ? [...allPaths].sort((a, b) => {
+              for (const field of opts.order ?? []) {
```

Comment:
oh, maybe create different sort functions for sort each strategy?

Reply:
I moved this to separated `getSwapPathComparator` and added unit tests for it, it's a generic function and strategy passed as priority list of liquidity and/or path length